OpenRA.Mods.Common/TraitsInterfaces.cs

```diff
@@ -442,4 +443,15 @@ public interface ITargetableCells
 
 	[RequireExplicitImplementation]
 	public interface IPreventsShroudReset { bool PreventShroudReset(Actor self); }
+
+	[RequireExplicitImplementation]
+	public interface IBotModuleInfo : ITraitInfoInterface { }
+
+	[RequireExplicitImplementation]
+	public interface IBotModule
+	{
+		string Name { get; }
+		void Activate(HackyAI ai);
```

Comment:
Summarising my comments from IRC for completeness: Ideally we could remove `HackyAI` here, or, even better, remove `Activate` completely and enable these modules using conditions.  This would then make it possible to enable these traits in campaign missions.

Reply:
The `HackyAI.QueueOrder` can move into the `IBot` interface in order to support this.

Reply:
As discussed on IRC I'm keeping this Activate for now, as it's the easiest way to pass `IBot` (and indirectly `Player`) to the module before anything else might need it.

Reply:
How about something like 186ecfa826d9ca92d314adfe6f2500e790782e1b?

Reply:
> Would something like this help: 186ecfa?

Yes, I think that's a good idea.